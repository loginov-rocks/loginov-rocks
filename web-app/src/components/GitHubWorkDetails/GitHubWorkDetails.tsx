import { GitHubRepo } from '@loginov-rocks/loginov-rocks-shared';
import * as React from 'react';

interface Props {
  repo: GitHubRepo;
}

export const GitHubWorkDetails: React.FC<Props> = ({ repo }) => (
  <>

    {repo.description && (
      <>
        {repo.description}
        <br />
      </>
    )}

    {repo.language && (
      <>
        <span title="Language">{repo.language}</span>
        {' '}
        /
        {' '}
      </>
    )}

    {repo.stars > 0 && (
      <>
        <span title="GitHub Stars">
          &#x2B50;
          {repo.stars}
        </span>
        {' '}
        /
        {' '}
      </>
    )}

    <span title="Last Updated">
      {new Date(repo.updatedAt).toLocaleDateString(undefined, {
        day: 'numeric',
        month: 'short',
        year: 'numeric',
      })}
    </span>

    {repo.latestTag && (
      <>
        {' '}
        /
        {' '}
        <span title="Latest Tag">{repo.latestTag}</span>
      </>
    )}

  </>
);
