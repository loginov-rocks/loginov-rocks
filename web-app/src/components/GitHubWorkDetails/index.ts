import { GitHubWorkDetails } from './GitHubWorkDetails';

export { GitHubWorkDetails };
