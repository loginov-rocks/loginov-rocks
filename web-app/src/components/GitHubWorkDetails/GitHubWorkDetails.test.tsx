/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { GitHubWorkDetails } from './GitHubWorkDetails';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const gitHubMockData = require('@loginov-rocks/loginov-rocks-shared/src/GitHub/__fixtures__/data.json');

it('matches snapshot', () => {
  const wrapper = shallow(
    <GitHubWorkDetails
      repo={gitHubMockData.repos[0]}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
