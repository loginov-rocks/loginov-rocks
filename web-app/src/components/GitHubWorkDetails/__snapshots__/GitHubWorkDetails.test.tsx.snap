// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  Clean but full featured AngularJS boilerplate using Gulp workflow and best practices
  <br />
  <span
    title="Language"
  >
    JavaScript
  </span>
   
  /
   
  <span
    title="GitHub Stars"
  >
    ‚≠ê
    29
  </span>
   
  /
   
  <span
    title="Last Updated"
  >
    Feb 25, 2021
  </span>
   
  /
   
  <span
    title="Latest Tag"
  >
    v1.0.10
  </span>
</Fragment>
`;
